cloud/services/container/nodepools/reconcile.go

```diff
@@ -354,6 +354,27 @@ func (s *Service) checkDiffAndPrepareUpdateVersionOrImage(existingNodePool *cont
 			Taints: desiredKubernetesTaints,
 		}
 	}
+	// Locations
+	desiredLocations := s.scope.GCPManagedMachinePool.Spec.NodeLocations
+	if !reflect.DeepEqual(desiredLocations, existingNodePool.Locations) {
```

Comment:
Can we change this to use cmp.Equals?

(and also the other instances below)

Reply:
Updated.